src/custom-elements/src/components/date-picker/date-picker.tsx

```diff
@@ -66,6 +68,47 @@ export class DatePicker {
     }
   }
 
+  _checkDateValidity(dateString) {
+    const parts = dateString.split("/");
+    const day = parseInt(parts[1], 10);
+    const month = parseInt(parts[0], 10);
+    const year = parseInt(parts[2], 10);
+    const monthLength = [ 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 ];
+
+    if(!/^\d{1,2}\/\d{1,2}\/\d{4}$/.test(dateString) ||
+      month < 1 || month > 12) {
+      return false;
+    }
+    if(year % 400 == 0 || (year % 100 != 0 && year % 4 == 0)) {
+      monthLength[1] = 29;
+    }
+
+    return day > 0 && day <= monthLength[month - 1];
+  }
+
+  _checkDate() {
+    const inputDate = new Date(this._input.value);
```

Comment:
Please, use Day.js for date objects. It will be easier to add support for locales.